httpbp/error_logger.go

```diff
@@ -0,0 +1,78 @@
+package httpbp
+
+import (
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/reddit/baseplate.go/internal/prometheusbpint"
+	"github.com/reddit/baseplate.go/randbp"
+)
+
+var httpServerLoggingCounter = promauto.With(prometheusbpint.GlobalRegistry).NewCounterVec(prometheus.CounterOpts{
+	Name: "httpbp_server_upstream_issue_logs_total",
+	Help: "Number of logs emitted by stdlib http server regarding an upstream issue",
+}, []string{"upstream_issue"})
+
+type issueConfig struct {
+	sampleRate float64
+	counter    prometheus.Counter
+}
+
+var supportedIDs = []int64{
+	25192,
+}
+
+// This is a special zapcore used by stdlib http server to handle error logging.
+type wrappedCore struct {
+	zapcore.Core
+
+	issues map[int64]issueConfig
+}
+
+func (w wrappedCore) Check(entry zapcore.Entry, ce *zapcore.CheckedEntry) *zapcore.CheckedEntry {
+	// Example message:
+	//     URL query contains semicolon, which is no longer a supported separator; parts of the query may be stripped when parsed; see golang.org/issue/25192
+	if strings.Contains(entry.Message, "golang.org/issue/25192") {
+		cfg := w.issues[25192]
```

Comment:
I don't think the map is really helping you here.  Since the error message is hard-coded and we are hard coding the supported IDs and so on, I would make a field for each one and just do
```
w.issue25192.Inc()
```

etc.